numpy/core/arrayprint.py

```diff
@@ -734,29 +734,51 @@ def __call__(self, x):
         else:
             return "%s" % x
 
+class BoolFormat(object):
+    def __init__(self, data, **kwargs):
+        # in legacy printing style, include a space before True except in 0d
+        if kwargs.get('legacy', False):
+            self.truestr = ' True' if data.shape != () else 'True'
+            return
+
+        # add an extra space so " True" and "False" have the same length and
+        # array elements align nicely when printed, but only for arrays with
+        # more than one element (0d and nd)
+        self.truestr = ' True' if data.size > 1 else 'True'
```

Comment:
I don't remember the rationale here - is there any reason not to change this to `.shape == ()`, now that we don't `ravel`?

That is, is it a deliberate choice to change to `array([True])` instead of `array([ True])` for anything but the scalar case?

I do vaguely recall deciding _not_ to use `not data.all()` as a condition here, and I think using `()` here would be consistent with that choice

Reply:
It is a deliberate choice, but debate is welcome.

It is so that the 1d array constructed as `np.array([True])` prints as `np.array([True], dtype=bool)` (without a space) and not `np.array([ True], dtype=bool)`. If we changed the test to `data.shape == ()` then this array would be printed with a space, which I thought was slightly ugly without any other benefit.

Perhaps, as you argue, the benefit would be consistency with longer boolean ndarrays, which print with a space, eg `array([ True,  True], dtype=bool)`. The spaces are added so that if a user prints out multiple boolean arrays on different lines of output, then the elements are aligned with each other across lines, for easy reading. But I didn't think this was useful if we are only printing size-1 arrays, since there is no reading problem.

To me the slight ugliness of the space outweighs the lack of consistency, but say the word and I can change it.

Reply:
I slightly prefer the consistency here - it's also one fewer case that users downstream would need to change their doctests for.

I'd say either wait for a third opinion to break the tie (@mhvk?), or change it so I can merge sooner. It'd be trivial to change this in a later PR anyway, as long as it still hits 1.14

Reply:
Hmm, tie-breaker, me? I think my preferred alternative was to omit the space for any case in which the array was all True, in analogy with the case where all numbers are positive...